pkg/tsdb/azuremonitor/httpclient.go

```diff
@@ -1,19 +1,26 @@
 package azuremonitor
 
 import (
+	"crypto/tls"
 	"fmt"
 	"net/http"
 
 	"github.com/grafana/grafana-azure-sdk-go/azcredentials"
 	"github.com/grafana/grafana-azure-sdk-go/azhttpclient"
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
 
-	"github.com/grafana/grafana/pkg/infra/httpclient"
 	"github.com/grafana/grafana/pkg/setting"
 	"github.com/grafana/grafana/pkg/tsdb/azuremonitor/types"
 )
 
-func newHTTPClient(route types.AzRoute, model types.DatasourceInfo, settings *backend.DataSourceInstanceSettings, cfg *setting.Cfg, clientProvider httpclient.Provider) (*http.Client, error) {
+type Provider interface {
+	New(...httpclient.Options) (*http.Client, error)
+	GetTransport(...httpclient.Options) (http.RoundTripper, error)
+	GetTLSConfig(...httpclient.Options) (*tls.Config, error)
+}
+
+func newHTTPClient(route types.AzRoute, model types.DatasourceInfo, settings *backend.DataSourceInstanceSettings, cfg *setting.Cfg, clientProvider Provider) (*http.Client, error) {
```

Comment:
This should be `httpclient.Provider`, I don't believe we should be redeclaring the interface

Reply:
`httpclient.Provider` is a struct and we want it to be an interface so we can mock a provider when writing tests :)

Reply:
I don't think it should be in this case because we wouldn't want to import the package in the test file just for the interface.